From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Mon, 10 Apr 2023 14:25 +0100

As buffer size in do_loop() is FIPS_RNG_BUFFER_SIZE, so the random_step
should not bigger. If random_step > FIPS_RNG_BUFFER_SIZE,
update_kernel_random() will not write any date.

Upstream PR #190, already merged in master.

---

diff -aur a/rngd.c b/rngd.c
--- a/rngd.c
+++ b/rngd.c
@@ -623,6 +623,8 @@
 	case 's':
 		if (sscanf(arg, "%i", &arguments->random_step) == 0)
 			argp_usage(state);
+		if (arguments->random_step > FIPS_RNG_BUFFER_SIZE || arguments->random_step < 0)
+			arguments->random_step = FIPS_RNG_BUFFER_SIZE;
 		break;
 	case 'W': {
 		int n;
